### DISCUSSION TITLE
[Bug]: Nested routes not working fine

### DISCUSSION THREAD

**Discussion by mbilesanmi** [remix-run/react-router#9376] [Q&A] [answered]
### What version of React Router are you using?

v6.4.1

### Steps to Reproduce

Create a `BrowserRouter` similar to the following

<img width="658" alt="image" src="https://user-images.githubusercontent.com/26195944/193121856-0afa8009-99f4-4e42-a6a3-2d8c0fc237a5.png">

Try to navigate from `root` to `blog` and that works fine. Navigating from `blog` to `blogPost` updates the url, but still renders `DeferredBlogPostsPage` component.

### Expected Behavior

Nesting should work fine regardless of how deep it is when `loaders` or any other props are used.

### Actual Behavior

[Code sandbox deploy](https://z5tuur-5173.preview.csb.app/)
[Code sandbox](https://codesandbox.io/p/github/mbilesanmi/react-router-6.4-test/draft/new-architecture)

Navigating to the [Blog](https://z5tuur-5173.preview.csb.app/blog) page works fine. However, clicking into a [blog-post](https://z5tuur-5173.preview.csb.app/blog/1) doesn't work. It updates the url, but doesn't render expected component.

**Comment by yunior0814** [comment, marked as answer]
@mbilesanmi I recommend to do something like this
  ```
<Route
        id="blog"
        path="blog"
        element={<Outlet  />}
      >
        {/* <Route index /> */}
         <Route
            index
            loader={deferredBlogPostsLoader}
            element={<DeferredBlogPostsPage />}
        />
        <Route
          id="blogPost"
          path=":id"
          element={<PostDetailPage />}
          loader={blogPostLoader}
        />
      </Route>
```

**Reply by mbilesanmi** [reply]
This was the approach I ended up going with, though it meant I needed to have a separate loader on the parent `Route`. I was trying to solve for using the nested routes and handlers to build breadCrumbs, and the nesting could go deeper. I had to create a `Layout` that then contained the `Outlet` for this purpose.

**Reply by mbilesanmi** [reply]
Thanks @yunior0814 for taking the time to have a look.
